harness/determined/pytorch/deepspeed/_mpu.py

```diff
@@ -0,0 +1,50 @@
+from dataclasses import dataclass
+
+from deepspeed.runtime.pipe import topology
+
+from determined import _core
+
+
+@dataclass
+class ModelParallelUnit:
+    """
+    This class contains the functions we expect in order to accurately carry out parallel training.
+    For custom model parallel training, you need to subclass and override the functions before
+    passing it to the :class:`~determined.pytorch.deepspeed.DeepSpeedTrialContext` by calling
+    ``context.wrap_mpu(mpu)``.
+    """
+
+    data_parallel_rank: int
+    data_parallel_world_size: int
+    # Whether the rank should return metrics from training and evaluation steps.
+    # This is usually true for ranks in the last stage of pipeline parallel or
+    # model parallel training.
+    should_report_metrics: bool = True
+    # Whether the rank should build a data loader.
+    # This is usually true for ranks in the first or last stage of
+    # pipeline parallel or model parallel training.
+    should_build_data_loader: bool = True
+
+
+def get_data_parallel_mpu(dist_context: _core.DistributedContext) -> ModelParallelUnit:
```

Comment:
this isn't really any sort of getter.  Can we name it `make_data_parallel_mpu()` instead?  Same below.